### PR TITLE
Add navigation picker for dashboards

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Add navigation picker for dashboards.
The picker is populated with all panels and dashboard views. A selector has been added to be available in `ha-form`.

https://user-images.githubusercontent.com/5878303/191325035-479999a2-4d66-4fe5-a17f-b1f10cf579df.mp4

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-navigation-picker.ts [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+import type { HaComboBox } from "./ha-combo-box";
+import "./ha-icon";
+
+type NavigationItem = {
+  path: string;
+  icon: string;
+  title: string;
+};
+
+const DEFAULT_ITEMS = [
```
Comment (balloob) [inline src/components/ha-navigation-picker.ts:23]: Instead of this, why not use hass.panels?

Comment (piitaya) [reply src/components/ha-navigation-picker.ts:23]: We can replace the `fetchDashboards` by `hass.panels`. So we just have to fetch the config for each panel starting with `lovelace`. Seems a good idea 🙂

File: src/panels/lovelace/components/hui-action-editor.ts [hunk 2]
```diff
@@ -89,14 +90,14 @@ export class HuiActionEditor extends LitElement {
       </div>
       ${this.config?.action === "navigate"
         ? html`
-            <ha-textfield
+            <ha-navigation-picker
+              .hass=${this.hass}
               label=${this.hass!.localize(
```
Comment (bramkragten) [inline src/panels/lovelace/components/hui-action-editor.ts:95]:
```suggestion
              .label=${this.hass!.localize(
```

File: src/components/ha-navigation-picker.ts [hunk 3]
```diff
@@ -104,47 +104,42 @@ export class HaNavigationPicker extends LitElement {
...
+      this.navigationItems.push(createPanelNavigationItem(this.hass!, panel));
+
+      const config = panelViewConfig.get(panel.id);
+
+      if (!config) continue;
+
+      const viewItems = config.views.map<NavigationItem>((view, index) =>
+        createViewNavigationItem(panel.url_path, view, index)
+      );
+      this.navigationItems = this.navigationItems.concat(viewItems);
```
Comment (bramkragten) [inline src/components/ha-navigation-picker.ts:142]:
```ts
      config.views.forEach<NavigationItem>((view, index) =>
        this.navigationItems.push(createViewNavigationItem(panel.url_path, view, index))
      );
```

File: src/components/ha-navigation-picker.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    }
+
+    this.navigationItemsLoaded = true;
```
Comment (bramkragten) [inline src/components/ha-navigation-picker.ts:145]: We should set this to true as soon as possible, so we won't run this twice when quickly opening and closing

**Review by bramkragten — APPROVED** [review]

**Pull request merged**